src/wp-includes/kses.php

```diff
@@ -981,7 +981,7 @@ function wp_kses_split( $content, $allowed_html, $allowed_protocols ) {
 	$pass_allowed_html      = $allowed_html;
 	$pass_allowed_protocols = $allowed_protocols;
 
-	return preg_replace_callback( '%(<!--.*?(-->|$))|(<[^>]*(>|$)|>)%', '_wp_kses_split_callback', $content );
+	return preg_replace_callback( '%<!--(?!#exec\s).*?(?:-->|$)|(<(?:[^<>]*|(?1))*+(?<!--)>)|[<>]%s', '_wp_kses_split_callback', $content );
```

Comment:
The code here was changed previously via [Trac ticket 61009](https://core.trac.wordpress.org/ticket/61009) and [its changeset 58418](https://core.trac.wordpress.org/changeset/58418).

This PR needs to rebase against the current `trunk` and then consideration given to the newer version of the code.

Pinging @dmsnell, who worked on and committed the changes.